pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AbstractSchema.java

```diff
@@ -74,14 +74,13 @@ public T decode(ByteBuf byteBuf, byte[] schemaVersion) {
      * @param schemaVersion the version
      * @return the schema at that specific version
      * @throws SchemaSerializationException in case of unknown schema version
-     * @throws NullPointerException in case of null schemaVersion
+     * @throws NullPointerException in case of null schemaVersion and supportSchemaVersioning is true
      */
     public Schema<?> atSchemaVersion(byte[] schemaVersion) throws SchemaSerializationException {
-        Objects.requireNonNull(schemaVersion);
         if (!supportSchemaVersioning()) {
             return this;
-        } else {
-            throw new SchemaSerializationException("Not implemented for " + this.getClass());
         }
+        Objects.requireNonNull(schemaVersion);
```

Comment:
Why do we need to check null for `schemaVersion` ?  It's very strange here and outside return Optional.
```
else if (schema instanceof AbstractSchema) {
            byte[] schemaVersion = getSchemaVersion();
            return Optional.of(((AbstractSchema<?>) schema)
                    .atSchemaVersion(schemaVersion));          
        }
```

Reply:
@congbobo184